tests/components/file_upload/conftest.py

```diff
@@ -0,0 +1,16 @@
+"""Fixtures for FileUpload integration."""
+from io import StringIO
+
+import pytest
+
+
+@pytest.fixture
+def large_file_io() -> StringIO:
+    """Generate a large file on the fly."""
+    file = StringIO("")
+    for _ in range(10000):
+        file.write(
+            "Home Assistant is awesome. Open source home automation that puts local control and privacy first."
+        )
+    file.seek(0)
+    return file
```

Comment:
Don't really need a fixture for this, it's only used once. You can do btw:
```suggestion
@pytest.fixture
def large_file_io() -> StringIO:
    """Generate a large file on the fly."""
    return StringIO(
            1000 *
            "Home Assistant is awesome. Open source home automation that puts local control and privacy first."
        )
```

I actually am thinking if we really need to write such a big file. Instead, can we patch out `ONE_MEGABYTE` and so read out smaller chunks to test out all the queue stuff ? That way we avoid memory jumping and huge I/O during tests

Reply:
Brilliant idea, that works flawlessly!